RecoBTag/Combined/src/HeavyIonCSVTagger.cc

```diff
@@ -80,26 +79,11 @@ float HeavyIonCSVTagger::discriminator(const TagInfoHelper &tagInfo) const {
   std::map<std::string, float> inputs;
   
   bool notTaggable = false;
-  bool noTrack = false;
-  float vtxMassVal = 0.;
-
-  for (auto &mva_var : variables_) {
-    //vectorial tagging variable
-    if (mva_var.has_index) {
-      std::vector<float> vals = vars.getList(mva_var.id, false);
-      inputs[mva_var.name] = (vals.size() > mva_var.index) ? vals[mva_var.index] : mva_var.default_value;
-      if (mva_var.name == "TagVarCSV_trackSip3dSig_0" && inputs[mva_var.name] < -98.999) noTrack = true;      
-    }
-    //single value tagging var
-    else {
-      inputs[mva_var.name] = vars.get(mva_var.id, mva_var.default_value);
-      //IK: vtxMass check to check vtxType: vtxType = 2 (no vtx), vtxMass < 0, vtxType = 1 (pseudo vtx), vtxMass > 0
-      if (mva_var.name == "TagVarCSV_vertexMass") vtxMassVal = inputs[mva_var.name];
-    }
-  }
+  std::vector<float> tagValList = vars.getList(reco::btau::trackSip3dSig, false);
+  bool noTrack = (tagValList.size() == 0);
+  bool noVertex = (vars.get(reco::btau::vertexCategory, -1.0) == 2); 
 
```

Comment:
Sorry if i was unclear, but you still need the for loop, it fills the input vector passed to the mva. It is just these noTrack/noVertex booleans that are used to check if the jet is taggable should be outside of it.

So  
```suggestion
  bool noVertex = (vars.get(reco::btau::vertexCategory, -1.0) == 2); 
  for (auto &mva_var : variables_) {
    //vectorial tagging variable
    if (mva_var.has_index) {
      std::vector<float> vals = vars.getList(mva_var.id, false);
      inputs[mva_var.name] = (vals.size() > mva_var.index) ? vals[mva_var.index] : mva_var.default_value;   
    }
    //single value tagging var
    else {
      inputs[mva_var.name] = vars.get(mva_var.id, mva_var.default_value);
    }
  }
```